soco/events.py

```diff
@@ -299,87 +142,68 @@ def run(self):
-        #: configured to listen.
-        # Empty for the moment. (It is set in `start`)
-        self.address = ()
 
-    def start(self, any_zone):
-        """Start the event listener listening on the local machine at port 1400
-        (default)
+    def listen(self, ip_address):
+        """Start the event listener listening on the local machine at
+        port 1400 (default).
+
+        Make sure that your firewall allows connections to this port.
 
-        Make sure that your firewall allows connections to this port
+        This method is called by `soco.events_base.EventListenerBase.start`
 
         Args:
-            any_zone (SoCo): Any Sonos device on the network. It does not
-                matter which device. It is used only to find a local IP address
-                reachable by the Sonos net.
+            ip_address (str): The local network interface on which the server
+                should start listening.
+        Returns:
+            int: `requested_port_number`. Included for
+                compatibility with `soco.events_twisted.EventListener.listen`
 
         Note:
             The port on which the event listener listens is configurable.
             See `config.EVENT_LISTENER_PORT`
         """
-
-        # Find our local network IP address which is accessible to the
-        # Sonos net, see http://stackoverflow.com/q/166506
-        with self._start_lock:
-            if not self.is_running:
-                # Use configured IP address if there is one, else detect
-                # automatically.
-                if config.EVENT_LISTENER_IP:
-                    ip_address = config.EVENT_LISTENER_IP
-                else:
-                    temp_sock = socket.socket(socket.AF_INET,
-                                              socket.SOCK_DGRAM)
-                    temp_sock.connect((any_zone.ip_address,
-                                       config.EVENT_LISTENER_PORT))
-                    ip_address = temp_sock.getsockname()[0]
-                    temp_sock.close()
-
-                # Start the event listener server in a separate thread.
-                self.address = (ip_address, config.EVENT_LISTENER_PORT)
-                self._listener_thread = EventServerThread(self.address)
-                self._listener_thread.daemon = True
-                self._listener_thread.start()
-                self.is_running = True
-                log.info("Event listener started")
-
-    def stop(self):
-        """Stop the event listener."""
+        address = (ip_address, self.requested_port_number)
+        self._listener_thread = EventServerThread(address)
+        self._listener_thread.daemon = True
+        self._listener_thread.start()
+        return self.requested_port_number
+
+    def stop_listening(self, address):
+        """Stop the listener."""
         # Signal the thread to stop before handling the next request
         self._listener_thread.stop_flag.set()
```

Comment:
Another suggestion: For the sake of encapsulation, could you create `stop()` methods for the threads, which just do this:
```python
def stop():
    self.stop_flag.set()
```